modules/gateway/nodes_test.go

```diff
@@ -222,9 +224,9 @@ func TestRelayNodes(t *testing.T) {
 
 	// g2 should have received g3's address from g1
```

Comment:
this comment is incorrect, g3 should have received g2's address from g1. Logic is correct.